Zend/zend_compile.c

```diff
@@ -6242,7 +6240,7 @@ static void zend_compile_foreach(zend_ast *ast) /* {{{ */
 
 static void zend_compile_if(zend_ast *ast) /* {{{ */
```

Comment:
same (and probably also elsewhere). Alternatively we could probably make `ast` cleanly `const` using `_Generic` by doing the following:

```
#define zend_ast_get_list(X) _Generic((X),     \
              const zend_ast*: zend_ast_get_list_const, \
                  default: zend_ast_get_list_mut,  \
              )(X)
```
or so.

Reply:
Yes, but they lose qualifiers, I haven't double-checked, but I'm fairly certain https://thephd.dev/improving-_generic-in-c2y goes into the nitty-gritty details.